### PR TITLE
Rank search results by levenshtein distance

### PR DESCRIPTION
**Pull request by Madoshakalaka**

### CODE REVIEW

**Comment by Madoshakalaka** [comment]
@eddieantonio what do you suggest we do to #113

File: CreeDictionary/utils/cree_lev_dist.py [hunk 1]
```diff
@@ -1,21 +1,56 @@
...
+
+
+def remove_cree_diacritics(input_str) -> str:
+    """    
+    >>> remove_cree_diacritics('ā')
+    'a'
+    >>> remove_cree_diacritics('â')
+    'a'
     """
+    nfkd_form = unicodedata.normalize("NFKD", input_str)
```
Comment (eddieantonio) [inline CreeDictionary/utils/cree_lev_dist.py]: Could you justify why you're using NFKD over NFD? For the expected inputs, they would both have the same effect.

**Comment by eddieantonio** [comment]
We can't really close #113 until we incorporate wordform weights for each entry, but at least ranking the results using edit distance gets us better weights

File: CreeDictionary/utils/cree_lev_dist.py (hunk 1 above)
Comment (Madoshakalaka) [reply CreeDictionary/utils/cree_lev_dist.py]: I googled "strip diacritics python" and this function is copy pasted. You can change it if neccesary.

File: CreeDictionary/utils/cree_lev_dist.py [hunk 2]
```diff
@@ -1,21 +1,56 @@
...
+    only_ascii = nfkd_form.encode("ASCII", "ignore").decode("utf-8")
```
[the remark this answers is not in the document]
Comment (Madoshakalaka) [reply CreeDictionary/utils/cree_lev_dist.py]: ikr

File: CreeDictionary/utils/cree_lev_dist.py (hunk 1 above)
Comment (eddieantonio) [reply CreeDictionary/utils/cree_lev_dist.py]: This is not a fool-proof code snippet, but I guess it works for now ¯\_(ツ)_/¯

**Comment by eddieantonio** [comment]
Does this close #96?

**Comment by Madoshakalaka** [comment]
Don't close it right now. I'm updating test examples from made-up ones to "acahkos", and just found the function returns not expected distance. I'm in the middle of debugging it.

**Comment by aarppe** [comment]
@Madoshakalaka @eddieantonio From the linguist's perspective on what they might want to tweak, regardless of the computational implemention, where are the weights explicitly specified? Cf. the proposed modified weights in the comment: https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/issues/96#issuecomment-549648171

**Comment by aarppe** [comment]
@Madoshakalaka for a comprehensive overview with more granularity as to how the edits should be weighted, see https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/issues/96#issuecomment-549648171

**Pull request merged**